doc/introduction/circuits.rst

```diff
@@ -241,9 +241,23 @@ a particular layout or topology. For example, the ``gbs`` compile target will
 compile a circuit consisting of Gaussian operations and Fock measurements
 into canonical Gaussian boson sampling form.
 
-.. code-block:: python3
+>>> prog2 = prog.compile('gbs')
+
+Programs can also be compiled for specific hardware devices using the Xanadu Cloud platform.
+After instantiating a remote engine for the target hardware, the device specifications
+can be accessed and used for compilation:
+
+>>> eng = sf.RemoteEngine("X8")
+>>> device = eng.device_spec
+>>> prog2 = prog.compile(device)
```

Comment:
An empty program (only measurements) seems to raise an error. Might be worth putting a check such that it's checked if `seq[0].op.p[0]` is going to be valid.
```python
import strawberryfields as sf
import strawberryfields.ops as ops

prog = sf.Program(8)

with prog.context as q:
    ops.MeasureFock() | q

eng = sf.RemoteEngine("X8")
device = eng.device_spec
prog2 = prog.compile(device)
```

Output:
```python
---> 11 prog2 = prog.compile(device)

~/xanadu/strawberryfields/strawberryfields/program.py in compile(self, device_or_compiler, force_compiler, **kwargs)
    553             seq = pu.optimize_circuit(seq)
    554 
--> 555         seq = compiler.compile(seq, self.register)
    556 
    557         # create the compiled Program

~/xanadu/strawberryfields/strawberryfields/compilers/xcov.py in compile(self, seq, registers)
     84 
     85         # extract the compiled symplectic matrix
---> 86         S = seq[0].op.p[0]
     87 
     88         if len(used_modes) != n_modes:

IndexError: list index out of range
```

Reply:
Good catch @antalszava! It looks like this is an existing bug in Xcov and Xunitary that are present in master, but only now being picked up since they are now the default.

Reply:
This has now been fixed as of 4487200 🙂